apps/opik-backend/src/main/java/com/comet/opik/domain/LlmProviderApiKeyService.java

```diff
@@ -40,21 +43,35 @@ class LlmProviderApiKeyServiceImpl implements LlmProviderApiKeyService {
     private final @NonNull TransactionTemplate template;
 
     @Override
-    public ProviderApiKey get(UUID id, String workspaceId) {
-        log.info("Getting provider api key with id '{}', workspaceId '{}'", id, workspaceId);
+    public ProviderApiKey find(UUID id, String workspaceId) {
 
         ProviderApiKey providerApiKey = template.inTransaction(READ_ONLY, handle -> {
 
             var repository = handle.attach(LlmProviderApiKeyDAO.class);
 
             return repository.fetch(id, workspaceId).orElseThrow(this::createNotFoundError);
         });
-        log.info("Got provider api key with id '{}', workspaceId '{}'", id, workspaceId);
 
         return providerApiKey.toBuilder()
                 .build();
     }
 
+    @Override
+    public Page<ProviderApiKey> find(String workspaceId) {
+        List<ProviderApiKey> providerApiKeys = template.inTransaction(READ_ONLY, handle -> {
+            var repository = handle.attach(LlmProviderApiKeyDAO.class);
+            return repository.find(workspaceId);
+        });
+
+        if (CollectionUtils.isEmpty(providerApiKeys)) {
+            return ProviderApiKey.ProviderApiKeyPage.empty(0);
+        }
+
+        return new ProviderApiKey.ProviderApiKeyPage(
+                0, providerApiKeys.size(), providerApiKeys.size(),
+                providerApiKeys, List.of());
```

Comment:
I imagine this will be changed in the future, right?

Reply:
Yes, right now we can have only one element here at max. 
Once we could actually have more than 1 page and have some sorting/filtering this will be updated.